src/overlays/actors/ovl_Bg_Tobira01/z_bg_tobira01.c

```diff
@@ -28,14 +28,78 @@ const ActorInit Bg_Tobira01_InitVars = {
+            CutsceneManager_Stop(CS_ID_GLOBAL_TALK);
+        } else if (CutsceneManager_IsNext(actorCutscene)) {
+            CutsceneManager_StartWithPlayerCs(actorCutscene, &this->dyna.actor);
+            SET_WEEKEVENTREG(WEEKEVENTREG_88_40);
+            this->cutscenePlayed = 0;
+        } else {
+            CutsceneManager_Queue(actorCutscene);
+        }
+    } else if (!(CHECK_WEEKEVENTREG(WEEKEVENTREG_88_40)) && (this->doorYPositionTick == 0) &&
+               (play->actorCtx.playerImpact.timer != 0) && (play->actorCtx.playerImpact.type == 0) &&
+               (SurfaceType_GetSceneExitIndex(&play->colCtx, player->actor.floorPoly, player->actor.floorBgId) == 6)) {
+        this->cutscenePlayed = 1;
+        this->unk16C = 0;
+    }
+
+    tempDoorYPosition = this->doorYPositionTick;
+    if (CHECK_WEEKEVENTREG(WEEKEVENTREG_88_40)) {
+        this->doorYPositionTick += 1;
+    } else {
+        this->doorYPositionTick -= 1;
+    }
+
+    this->doorYPositionTick = CLAMP(this->doorYPositionTick, 0, 60);
+    if (tempDoorYPosition != this->doorYPositionTick) {
+        Actor_PlaySfx(&this->dyna.actor, NA_SE_EV_STONEDOOR_OPEN_S - SFX_FLAG);
+        this->dyna.actor.world.pos.y =
+            (this->yPosition = (this->doorYPositionTick * (5.0f / 3.0f)) + this->dyna.actor.home.pos.y);
+        this->isNight = 0xB4;
+    }
+
+    if (!(player->stateFlags1 & PLAYER_STATE1_40) && (CHECK_WEEKEVENTREG(WEEKEVENTREG_88_40)) &&
+        DECR(this->isNight) == 0) {
+        CLEAR_WEEKEVENTREG(WEEKEVENTREG_88_40);
+    }
+}
+
+void BgTobira01_Init(Actor* thisx, PlayState* play) {
+    BgTobira01* this = THIS;
+
+    DynaPolyActor_Init(&this->dyna, 1);
+    DynaPolyActor_LoadMesh(play, &this->dyna, &gGoronDoorCol);
+    CLEAR_WEEKEVENTREG(WEEKEVENTREG_88_40);
+    Actor_SetScale(&this->dyna.actor, 1.0f);
+    this->isNight = gSaveContext.save.isNight;
+    this->doorYPositionTick = 0;
+    this->actionFunction = BgTobira01_Action;
+}
+
+void BgTobira01_Destroy(Actor* thisx, PlayState* play) {
+    BgTobira01* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Tobira01/BgTobira01_Init.s")
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Tobira01/BgTobira01_Destroy.s")
+void BgTobira01_Update(Actor* thisx, PlayState* play) {
+    BgTobira01* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Tobira01/BgTobira01_Update.s")
+    this->actionFunction(this, play);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Tobira01/BgTobira01_Draw.s")
+void BgTobira01_Draw(Actor* thisx, PlayState* play) {
+    OPEN_DISPS(play->state.gfxCtx);
+    func_8012C28C(play->state.gfxCtx);
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_OPA_DISP++, gGoronDoorDL);
+    CLOSE_DISPS(play->state.gfxCtx);
+}
```

Comment:
```suggestion
void BgTobira01_Draw(Actor* thisx, PlayState* play) {
    OPEN_DISPS(play->state.gfxCtx);

    func_8012C28C(play->state.gfxCtx);
    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
    gSPDisplayList(POLY_OPA_DISP++, gGoronDoorDL);

    CLOSE_DISPS(play->state.gfxCtx);
}
```
Just a styling suggestion, we tend to put spaces around OPEN/CLOSE DISPS